syntax = "proto2";

message Request {
    required string inputMethod = 1;
    required string inputMethodName = 2;
    required string solutionMethod = 3;
    repeated string inputs = 4;
    optional int64  timeout = 5;
}

message Response {

    enum RunResultType {
        Success = 0;
        CompilerError = 1;
    }

    required RunResultType overallResultType = 1;
    repeated Output results = 2;

}

enum OutputResultType {
    Success = 0;
    RuntimeError = 1;
    SecurityError = 2;
    CompilerError = 3;
    TimeoutError = 4;
}

message Output {


    required string solutionOutput = 1;
    required OutputResultType solutionOutputType = 2;
    required string methodOutput = 3;
    required OutputResultType methodOutputType = 4;
    required bool match = 5;

}
